File: app/backend/internal/models/image.go
package models

import "time"

// Image represents uploaded medical image metadata
type Image struct {
	ID               uint        `gorm:"primaryKey;autoIncrement" json:"id"`
	SessionID        uint        `gorm:"not null;index" json:"session_id"`
	Session          ScanSession `gorm:"foreignKey:SessionID;constraint:OnDelete:CASCADE;" json:"session"`
	FileName         string      `gorm:"type:varchar(255);not null" json:"file_name"`
	FileURL          string      `gorm:"type:varchar(500);not null" json:"file_url"`
	FileSize         *int64      `json:"file_size"`
	MimeType         *string     `gorm:"type:varchar(100)" json:"mime_type"`
	DiagnosticResult *string     `gorm:"type:text" json:"diagnostic_result"`
	UploadedBy       *uint       `json:"uploaded_by"`
	Uploader         *User       `gorm:"foreignKey:UploadedBy;constraint:OnDelete:SET NULL;" json:"uploader"`
	CreatedAt        time.Time   `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt        time.Time   `gorm:"autoUpdateTime" json:"updated_at"`
}

File: app/frontend/src/types/image.ts
export interface MedicalImage {
  id: number;
  session_id: number;
  file_name: string;
  file_url: string;
  file_size?: number;
  mime_type?: string;
  diagnostic_result?: string | null;
  uploaded_by?: number;
  uploader_name?: string;
  created_at: string;
  updated_at?: string;
}

export interface ImageListResponse {
  data: MedicalImage[];
}

export interface UploadImageResponse {
  message: string;
  image: MedicalImage;
}

export interface DeleteImageResponse {
  message: string;
}
